### DISCUSSION TITLE
Error HH1001: Invalid source name

### DISCUSSION THREAD

**Discussion by deegrate** [NomicFoundation/hardhat#4045] [Troubleshooting] [answered]
I am working on customizing a crypto arbitrage bot with flash loans and have pushed the program to a repository here: 
[https://github.com/deegrate/AutoArb2.git]

Dependencies:
- Solidity (Writing Smart Contract)
- Javascript (React & Testing)
- [Hardhat](https://hardhat.org/) (Development Framework)
- [Ethers.js](https://docs.ethers.io/v5/) (Blockchain Interaction)
- [Alchemy](https://www.alchemy.com/) (Blockchain Connection)

I have only been able to input 3 commands before getting stuck.
1. "npm install"
2. "npx hardhat node"
3. "npx hardhat run scripts\deploy.js --network localhost"

Then, I get this error:

```
Error HH1001: Invalid source name .. Expected source name but found a relative path.
HardhatError: HH1001: Invalid source name .. Expected source name but found a relative path.
    at validateSourceNameFormat (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/utils/source-names.ts:27:11)
    at Resolver.resolveSourceName (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/internal/solidity/resolver.ts:89:29)
    at Resolver.resolveImport (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/internal/solidity/resolver.ts:174:35)
    at /mnt/c/pr0/AutoArb/node_modules/hardhat/src/internal/solidity/dependencyGraph.ts:176:28
    at async Promise.all (index 2)
    at DependencyGraph._addDependenciesFrom (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/internal/solidity/dependencyGraph.ts:174:5)
    at async Promise.all (index 0)
    at Function.createFromResolvedFiles (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/internal/solidity/dependencyGraph.ts:13:5)
    at SimpleTaskDefinition.action (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/builtin-tasks/compile.ts:204:31)
    at Environment._runTaskDefinition (/mnt/c/pr0/AutoArb/node_modules/hardhat/src/internal/core/runtime-environment.ts:330:14)
```
	
I've been all through the code searching for a relative path that should be a source name, I've tried Google, StackOverflow, StackExchange, Alchemy, and Hardhat. Any help moving me further along in the process would be hugely appreciated.
Can anyone help me clear this error?

**Comment by fvictorio** [comment]
@deegrate your repo has some `:Zone.Identifier` files, maybe that's causing the issue?

Hardhat should ignore those files though. If you can confirm that deleting them makes the problem go away, please open an issue about Hardhat not working correctly when there are [Alternate Data Streams](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/e2b19412-a925-4360-b009-86e3b8a020c8) in the contracts directory.

**Reply by deegrate** [reply]
I beleive those files only showed up after I pushed the project to a Github repository. They don't exist in the root folder (or at least they don't show in Windows Explorer 11). I deleted them from the repository, but there were none to delete in the local project folder. Any other ideas?

**Reply by fvictorio** [reply]
Can you run `npx hardhat console` in your project and then this?

```
await hre.run("compile:solidity:get-source-paths")
```

What does that return?

**Reply by deegrate** [reply]
npx hardhat console returns the same error: "Error HH1001: Invalid source name .. Expected source name but found a relative path."

I tried the second command anyway, and got this: " -bash: syntax error near unexpected token `(' "

**Reply by fvictorio** [reply]
Sorry, use `npx hardhat console --no-compile`

**Reply by deegrate** [reply]
Ok, I input: "npx hardhat console --no-compile", and it opened an instance of the console...

Welcome to Node.js v18.12.0.
Type ".help" for more information.
>

I tried rephrasing my deployment command to more resemble HardHat docs like this:
"npx hardhat run --network localhost scripts/deploy.js"
...and received the same HH1001 error.

**Reply by fvictorio** [reply]
> Ok, I input: "npx hardhat console --no-compile", and it opened an instance of the console...

Yes, and after that run `await hre.run("compile:solidity:get-source-paths")` as I explained before, and paste here the result.

**Reply by deegrate** [reply]
Apologies... here is the return:

`> await hre.run("compile:solidity:get-source-paths")
[
  '/mnt/c/pr0/AutoArb/contracts/Arbitrage.sol',
  '/mnt/c/pr0/AutoArb/contracts/Lock.sol'
]
>`

**Reply by deegrate** [reply]
Ahhhh... here's my thought after your help:

These are the correct source paths I need, so if I edit the corresponding relative paths, then that should clear the error. Am I on the right track?

**Reply by fvictorio** [reply, marked as answer]
Uhm, I don't think so. I just wanted to see if the ADS files were showing up in that list, but they aren't.

I'm not sure what can be going on. I couldn't reproduce this on linux. Some things you can try are:

1. Run `npx hardhat clean` and then `npx hardhat compile`
2. Clone the repo from scratch in a different directory and try again

**Reply by deegrate** [reply]
Yep... I think that is the road to take at this point. Thanks so much for your help. Hugely appreciated.

**Reply by deegrate** [reply]
Started from scratch and ran a successful test. Thanks again.
